src/pe/characteristic.rs

```diff
@@ -90,10 +92,64 @@ pub const IMAGE_FILE_DLL: u16 = 0x2000;
 pub const IMAGE_FILE_UP_SYSTEM_ONLY: u16 = 0x4000;
 pub const IMAGE_FILE_BYTES_REVERSED_HI: u16 = 0x8000;
 
+fn has_flag(characteristics: u16, flag: u16) -> bool {
+    characteristics & flag == flag
+}
+
 pub fn is_dll(characteristics: u16) -> bool {
-    characteristics & IMAGE_FILE_DLL == IMAGE_FILE_DLL
+    has_flag(characteristics, IMAGE_FILE_DLL)
 }
 
 pub fn is_exe(characteristics: u16) -> bool {
-    characteristics & IMAGE_FILE_EXECUTABLE_IMAGE == IMAGE_FILE_EXECUTABLE_IMAGE
+    has_flag(characteristics, IMAGE_FILE_EXECUTABLE_IMAGE)
+}
+
+/// Validate the characteristics.
+///
+/// `is_image` being `true` indicates the characteristics is in a `PE`,
+/// being `false` indicates the characteristics is in a `Coff`.
+pub fn validate(characteristics: u16, is_image: bool) -> error::Result<()> {
```

Comment:
ditto same issue with boolean blindness here.

Also does this really need to be `pub`; I don't see many people calling this.